HWM-51: added push to ecr stage

gregwolversonHO opened the pull request:
## What? 

Update the drone pipeline to add a push_to_ecr stage which will run on any push to a feature or the main branch and any pull_request.

## Why? 

So we can build and push images into ECR, especially on feature branches (where we can subsequently deploy to branch). 

## How? 
## Testing?
## Screenshots (optional)
## Anything Else? (optional)
## Check list

- [X] I have reviewed my own pull request for linting issues (e.g. adding new lines)
- [ ] I have written tests (if relevant)
- [ ] I have created a JIRA number for my branch
- [ ] I have created a JIRA number for my commit
- [ ] I have followed the chris beams method for my commit https://cbea.ms/git-commit/
here is an [example commit](https://github.com/UKHomeOfficeForms/hof/commit/810959f391187c7c4af6db262bcd143b50093a6e)
- [ ] Ensure drone builds are green especially tests
- [ ] I will squash the commits before merging

gregwolversonHO committed:
HWM-51: added push to ecr stage

gregwolversonHO committed:
HWM-51: fixing sonar tests location

gregwolversonHO committed:
HWM-51: fixing sonar tests location

gregwolversonHO committed:
HWM-51: updating ecr push to only happen on main

gregwolversonHO committed:
HWM-51: updating sonar settings and package

adityababumallisettiHO commented on .drone.yaml:
```diff
@@ -60,23 +60,28 @@ steps:
+      repo: so/webmessengers
+      registry: 340268328991.dkr.ecr.eu-west-2.amazonaws.com
+      tags:
+        - latest
```
In scenarios where both pull requests and direct pushes are made to the main or master branch, triggering a Docker image build can lead to the latest tag being overwritten by whichever build runs last. To avoid this, it's recommended to use the Drone commit SHA as a unique image tag instead.

gregwolversonHO replied:
I can remove the `-latest` and just use the DRONE_COMMIT for the tag

adityababumallisettiHO commented on .drone.yaml:
```diff
@@ -60,23 +60,28 @@ steps:
+    when:
+      branch: main
+      event:
+      - push
```
To ensure this step runs for new pull requests, we may want to configure the pipeline to trigger on the pull_request event in Drone. This allows the build to execute even before changes are merged into the main or master branch.

gregwolversonHO replied:
We don't want to push to ECR on branch changes though? Only when the code is merged to main? Or am I getting confused about your comment?

adityababumallisettiHO replied:
Usually When we create a PR, Using the conditions we set in Drone step, It triggers the builds and deploys in Branch namespace. Once we test and confirm, we then merge (push) to Main/Master branch to deploy in UAT and STAGE. We build image everytime before deploying to Branch, UAT and Stage. This is only way to test the Docker image in branch namespace. As we know Docker image can change time to time depending on the inclusion and exclusion of dependencies/artifacts in the Package.json

gregwolversonHO replied:
Do you have an example of this that I can use for the `when` step here please? Might be easier than me trying to work it out

adityababumallisettiHO replied:
This is the latest pipeline. We can configure pipeline based on this project. Feel free to customize as per your requirements. 
https://github.com/UKHomeOffice/controlled-substance-licence/blob/master/.drone.yml

gregwolversonHO replied:
I've updated the push config based on CSL now

TemitopeAyokuHO committed:
HWM-52 Add gov uk css (#4)

- Added necessary files and folders from gov frontend node module

- Installed gov frontend module

- Added configuration setting

gregwolversonHO committed:
HWM-51: updating the push to ecr to work on feature and main

gregwolversonHO committed:
HWM-51: updating the trigger on builds

gregwolversonHO committed:
HWM-51: adding trigger to scan and build steps

adityababumallisettiHO reviewed (approved):
lgtm
